### ISSUE TITLE
Space lyrics at end of measure

### ISSUE THREAD

**Issue by sschmidTU** [opensheetmusicdisplay/opensheetmusicdisplay#324]
At the moment, we cannot avoid lyrics collisions at the end of the measure, because stretching the next measure does not change the position of the first note. The first note is always at x=0.
A good test case for this is the Austrian National Anthem, which has many long lyrics at the end of a measure:
![image](https://user-images.githubusercontent.com/33069673/44107616-4319b64c-9ff8-11e8-9268-12ab78551822.png)
This is the only part of the piece and the only one of our sample files that still has lyrics collisions.

The problem is not the multiple verses, we already handle each verse separately in #320:
![image](https://user-images.githubusercontent.com/33069673/44107718-9c252bc2-9ff8-11e8-84ba-e61cfd854ae9.png)

I have a solution in mind that could be a little tricky for our code structure. We could stretch the previous measure if we find a collision in the first lyric of the measure. The only decent solution i can think of is to let calculateMeasureWidthFromLyrics return not just the new width but an object with the width and an optional parameter that tells MusicSheetCalculator to stretch the previous measure. It's not the most elegant solution, but the alternatives might be uglier.
``` for (let i: number = 1; i < this.graphicalMusicSheet.MeasureList.length; i++) {
                measures = this.graphicalMusicSheet.MeasureList[i];
                minimumStaffEntriesWidth = this.calculateMeasureXLayout(measures);
                minimumStaffEntriesWidth = this.calculateMeasureWidthFromLyrics(measures, minimumStaffEntriesWidth);
                // instead, do this:
                // widthAndStretchPrevious = this.calculateMeasureWidthFromLyrics(...)
                // if (widthAndStretch.stretchPrevious { this.graphicalMusicSheet.MeasureList[i-1].setStretch... }
                MusicSheetCalculator.setMeasuresMinStaffEntriesWidth(measures, minimumStaffEntriesWidth);
                minLength = Math.max(minLength, minimumStaffEntriesWidth * 1.2 + maxInstructionsLength);
            }

**Comment by matt-uib** [comment]
Actually here the left-aligned lyrics positioning could be a good solution:
If every syllable is only reaching out to the right of the notes, you can handle the problem again only within the actual measure and don't have to look into the next one.
You could even take the constant space at the beginning of each measure into account and use it as the final border where the last syllable should not go beyond. So take actual measure size + const begin measure space (coming from the next measure).
(By shilfting the lyrics to left-aligned, the overal space demand stays the same)

**Comment by sschmidTU** [comment]
Fixed in #356 by left-aligning lyrics as matt-uib suggested, and spacing towards end of measure with a small overlap into next measure.
